src/game_interpreter_battle.cpp

```diff
@@ -732,12 +732,125 @@ bool Game_Interpreter_Battle::CommandManiacChangeBattleCommandEx(lcf::rpg::Event
 	return true;
 }
 
-bool Game_Interpreter_Battle::CommandManiacGetBattleInfo(lcf::rpg::EventCommand const&) {
+bool Game_Interpreter_Battle::CommandManiacGetBattleInfo(lcf::rpg::EventCommand const& com) {
 	if (!Player::IsPatchManiac()) {
 		return true;
 	}
 
-	Output::Warning("Maniac Patch: Command GetBattleInfo not supported");
+	int target_flags = com.parameters[0];
+	int target_reference_flags = com.parameters[2];
+	int target_reference_identifier = com.parameters[3];
+	int information_flags = com.parameters[1];
+	int information_identifier = com.parameters[4];
+
+	int target_id = ValueOrVariable(target_reference_flags, target_reference_identifier);
+
+	auto executeOperationSingle = [information_flags, information_identifier, target_id](Game_Battler* battler, TargetType battler_type) {
+		if (!battler) {
+			MissingTargetWarning("CommandManiacGetBattleInfo", battler_type, target_id);
+			return;
+		}
+
+		switch (information_flags)
+		{
+			case 0:
+				// parameter buffs: attack, defense, mind, agility
+				Main_Data::game_variables->Set(information_identifier, battler->GetAtkModifier());
+				Main_Data::game_variables->Set(information_identifier + 1, battler->GetDefModifier());
+				Main_Data::game_variables->Set(information_identifier + 2, battler->GetSpiModifier());
+				Main_Data::game_variables->Set(information_identifier + 3, battler->GetAgiModifier());
+				break;
+			case 1:
+			{
+				// states: size, [...state_id]
+				auto states = lcf::Data::states.size();
+				Main_Data::game_variables->Set(information_identifier, states);
+				for (size_t i = 0; i < states; i++)
+				{
+					Main_Data::game_variables->Set(information_identifier + i + 1, battler->HasState(lcf::Data::states.at(i).ID));
+				}
+				break;
+			}
+			case 2:
+			{
+				// elements: size, [...element_id]
+				auto elements = lcf::Data::attributes.size();
+				Main_Data::game_variables->Set(information_identifier, elements);
+				for (size_t i = 0; i < elements; i++)
+				{
+					Main_Data::game_variables->Set(information_identifier + i + 1, battler->GetAttributeRateShift(lcf::Data::attributes.at(i).ID) + 1);
+				}
+				break;
+				break;
```

Comment:
This command looks also straight forward 👍. There are two ``break`` here and instead of ``at()`` you can use the index ``[]``. Besides this have no complaints here. Thanks

Reply:
Done and done!